--- irssi-0.8.6/irssi.conf.orig	Thu Sep 12 01:00:08 2002
+++ irssi-0.8.6/irssi.conf	Tue Nov 19 14:47:41 2002
@@ -22,6 +22,7 @@
   },
   { address = "irc.ptnet.org"; chatnet = "PTnet"; port = "6667"; },
   { address = "irc.azzurra.org"; chatnet = "AzzurraNET"; port = "6667"; },
+  { address = "irc.pld.org.pl"; chatnet = "PLDNet"; port = "6667"; },
   { address = "silc.pspt.fi"; chatnet = "SILC"; port = "706"; }
 );
 
@@ -104,11 +105,20 @@
     max_msgs = "3";
   };
   SILC = { type = "SILC"; };
+  PLDNet = {
+    type = "IRC";
+    max_kicks = "4";
+    max_modes = "4";
+    max_msgs = "5";
+    max_whois = "4";
+    max_query_chans = "5";
+  };
 };
 
 channels = (
   { name = "#irssi"; chatnet = "ircnet"; autojoin = "No"; },
   { name = "#irssi"; chatnet = "freenode"; autojoin = "No"; },
+  { name = "#pldhelp"; chatnet = "PLDNet"; autojoin = "No"; },
   { name = "#silc"; chatnet = "silc"; autojoin = "No"; }
 );
 
