diff -ur irssi-0.8.10-rc5-o/configure.in irssi-0.8.10-rc5/configure.in
--- irssi-0.8.10-rc5-o/configure.in	2004-09-15 06:15:05.000000000 -0600
+++ irssi-0.8.10-rc5/configure.in	2005-05-29 01:32:38.000000000 -0600
@@ -236,6 +236,10 @@
     ;;
 esac
 
+AC_CHECK_LIB(popt, poptGetInvocationName, [
+	LIBS="$LIBS -lpopt"
+])
+
 
 AC_CHECK_FUNCS(mkfifo fcntl nl_langinfo)
 
@@ -351,7 +355,7 @@
     AC_MSG_RESULT([yes, using it])
 
     dnl * we have to do this at this point so we know what libs gmodule needs
-    if test ! -f $GLIB_DIR/.libs/libglib.a; then
+    if test ! -f $GLIB_DIR/.libs/libglib.la; then
       echo
       echo "configuring GLib ..."
       echo
@@ -745,21 +749,21 @@
 			dnl * building with static perl support
 			dnl * all PERL_LDFLAGS linking is done in fe-text
 			PERL_LINK_FLAGS="$PERL_LDFLAGS"
-			PERL_LINK_LIBS="../perl/libperl_core_static.la"
-			PERL_FE_LINK_LIBS="../perl/libfe_perl_static.la"
+			PERL_LINK_LIBS="../perl/libirssi_perl_core_static.la"
+			PERL_FE_LINK_LIBS="../perl/libirssife_perl_static.la"
 			PERL_LDFLAGS=
 			AC_DEFINE(HAVE_STATIC_PERL)
 
 			dnl * build only static library of perl module
 			perl_module_lib=
 			perl_module_fe_lib=
-			perl_static_lib=libperl_core_static.la
-			perl_static_fe_lib=libfe_perl_static.la
+			perl_static_lib=libirssi_perl_core_static.la
+			perl_static_fe_lib=libirssife_perl_static.la
 			PERL_LIBTOOL='$(SHELL) $(top_builddir)/libtool'
 		else
 			dnl * build dynamic library of perl module
-			perl_module_lib=libperl_core.la
-			perl_module_fe_lib=libfe_perl.la
+			perl_module_lib=libirssi_perl_core.la
+			perl_module_fe_lib=libirssife_perl.la
 			perl_static_lib=
 			perl_static_fe_lib=
 			PERL_LIBTOOL='$(SHELL) $(top_builddir)/libtool'
@@ -833,7 +837,7 @@
 AC_SUBST(CHAT_MODULES)
 AC_SUBST(irc_MODULES)
 
-CORE_LIBS="../core/libcore.a ../lib-config/libirssi_config.a ../lib-popt/libpopt.a"
+CORE_LIBS="../core/libirssi_core.la ../lib-config/libirssi_config.la"
 FE_COMMON_LIBS=""
 
 CHAT_LIBS=""
@@ -842,16 +846,16 @@
 	module_deinits=""
 	fe_module_inits=""
 	fe_module_deinits=""
-	CHAT_LIBS="$CHAT_LIBS ../$c/lib$c.a ../$c/core/lib${c}_core.a"
+	CHAT_LIBS="$CHAT_LIBS ../$c/libirssi_$c.la ../$c/core/libirssi_${c}_core.la"
 	if test -f $srcdir/src/fe-common/$c/module.h; then
-		FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/$c/libfe_common_$c.a "
+		FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/$c/libirssife_common_$c.la "
 	fi
 	for s in `eval echo \\$${c}_MODULES`; do
-		CHAT_LIBS="$CHAT_LIBS ../$c/$s/lib${c}_$s.a"
+		CHAT_LIBS="$CHAT_LIBS ../$c/$s/libirssi_${c}_$s.la"
 		module_inits="$module_inits ${c}_${s}_init();"
 		module_deinits="${c}_${s}_deinit(); $module_deinits"
 		if test -f $srcdir/src/fe-common/$c/$s/module.h; then
-			FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/$c/$s/libfe_${c}_$s.a "
+			FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/$c/$s/libirssife_${c}_$s.la "
 			fe_module_inits="$fe_module_inits fe_${c}_${s}_init();"
 			fe_module_deinits="fe_${c}_${s}_deinit(); $fe_module_deinits"
 		fi
@@ -879,7 +883,7 @@
 	fi
 done
 
-FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/core/libfe_common_core.a"
+FE_COMMON_LIBS="$FE_COMMON_LIBS../fe-common/core/libirssife_common_core.la"
 
 dnl ** common libraries needed by frontends
 COMMON_NOUI_LIBS="$CHAT_LIBS $CORE_LIBS"
diff -ur irssi-0.8.10-rc5-o/src/core/Makefile.am irssi-0.8.10-rc5/src/core/Makefile.am
--- irssi-0.8.10-rc5-o/src/core/Makefile.am	2004-08-19 18:10:01.000000000 -0600
+++ irssi-0.8.10-rc5/src/core/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,6 @@
-noinst_LIBRARIES = libcore.a
+libirssi_core_LTLIBRARIES = libirssi_core.la
+
+libirssi_coredir = /usr/lib
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -7,7 +9,7 @@
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
 	-DMODULEDIR=\""$(libdir)/irssi/modules"\"
 
-libcore_a_SOURCES = \
+libirssi_core_la_SOURCES = \
 	args.c \
 	channels.c \
 	channels-setup.c \
diff -ur irssi-0.8.10-rc5-o/src/fe-common/core/Makefile.am irssi-0.8.10-rc5/src/fe-common/core/Makefile.am
--- irssi-0.8.10-rc5-o/src/fe-common/core/Makefile.am	2004-09-15 06:15:05.000000000 -0600
+++ irssi-0.8.10-rc5/src/fe-common/core/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libfe_common_core.a
+libirssife_common_core_LTLIBRARIES = libirssife_common_core.la
+libirssife_common_coredir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src -I$(top_srcdir)/src/core/ \
@@ -6,7 +7,7 @@
 	-DHELPDIR=\""$(datadir)/irssi/help"\" \
 	-DTHEMESDIR=\""$(datadir)/irssi/themes"\"
 
-libfe_common_core_a_SOURCES = \
+libirssife_common_core_la_SOURCES = \
 	autorun.c \
 	chat-completion.c \
 	command-history.c \
diff -ur irssi-0.8.10-rc5-o/src/fe-common/irc/dcc/Makefile.am irssi-0.8.10-rc5/src/fe-common/irc/dcc/Makefile.am
--- irssi-0.8.10-rc5-o/src/fe-common/irc/dcc/Makefile.am	2004-05-10 14:45:03.000000000 -0600
+++ irssi-0.8.10-rc5/src/fe-common/irc/dcc/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libfe_irc_dcc.a
+libirssife_irc_dcc_LTLIBRARIES = libirssife_irc_dcc.la
+libirssife_irc_dccdir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -10,7 +11,7 @@
 	-DHELPDIR=\""$(datadir)/irssi/help"\" \
 	-DSYSCONFDIR=\""$(sysconfdir)"\"
 
-libfe_irc_dcc_a_SOURCES = \
+libirssife_irc_dcc_la_SOURCES = \
 	fe-dcc.c \
 	fe-dcc-chat.c \
 	fe-dcc-chat-messages.c \
diff -ur irssi-0.8.10-rc5-o/src/fe-common/irc/Makefile.am irssi-0.8.10-rc5/src/fe-common/irc/Makefile.am
--- irssi-0.8.10-rc5-o/src/fe-common/irc/Makefile.am	2004-07-11 09:45:03.000000000 -0600
+++ irssi-0.8.10-rc5/src/fe-common/irc/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,6 +1,7 @@
 SUBDIRS = dcc notifylist
 
-noinst_LIBRARIES = libfe_common_irc.a
+libirssife_common_irc_LTLIBRARIES = libirssife_common_irc.la
+libirssife_common_ircdir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -29,7 +30,7 @@
 	irc-completion.c \
 	module-formats.c 
 
-libfe_common_irc_a_SOURCES = \
+libirssife_common_irc_la_SOURCES = \
 	$(real_sources) \
         irc-modules.c
 
diff -ur irssi-0.8.10-rc5-o/src/fe-common/irc/notifylist/Makefile.am irssi-0.8.10-rc5/src/fe-common/irc/notifylist/Makefile.am
--- irssi-0.8.10-rc5-o/src/fe-common/irc/notifylist/Makefile.am	2004-05-10 14:45:03.000000000 -0600
+++ irssi-0.8.10-rc5/src/fe-common/irc/notifylist/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libfe_irc_notifylist.a
+libirssife_irc_notifylist_LTLIBRARIES = libirssife_irc_notifylist.la
+libirssife_irc_notifylistdir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -9,7 +10,7 @@
 	-DHELPDIR=\""$(datadir)/irssi/help"\" \
 	-DSYSCONFDIR=\""$(sysconfdir)"\"
 
-libfe_irc_notifylist_a_SOURCES = \
+libirssife_irc_notifylist_la_SOURCES = \
 	fe-notifylist.c \
 	module-formats.c
 
diff -ur irssi-0.8.10-rc5-o/src/irc/core/Makefile.am irssi-0.8.10-rc5/src/irc/core/Makefile.am
--- irssi-0.8.10-rc5-o/src/irc/core/Makefile.am	2004-05-10 14:45:04.000000000 -0600
+++ irssi-0.8.10-rc5/src/irc/core/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libirc_core.a
+libirssi_irc_core_LTLIBRARIES = libirssi_irc_core.la
+libirssi_irc_coredir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -6,7 +7,7 @@
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
 	$(GLIB_CFLAGS)
 
-libirc_core_a_SOURCES = \
+libirssi_irc_core_la_SOURCES = \
 	bans.c \
         ctcp.c \
         channels-query.c \
diff -ur irssi-0.8.10-rc5-o/src/irc/dcc/Makefile.am irssi-0.8.10-rc5/src/irc/dcc/Makefile.am
--- irssi-0.8.10-rc5-o/src/irc/dcc/Makefile.am	2004-05-10 14:45:04.000000000 -0600
+++ irssi-0.8.10-rc5/src/irc/dcc/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libirc_dcc.a
+libirssi_irc_dcc_LTLIBRARIES = libirssi_irc_dcc.la
+libirssi_irc_dccdir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -6,7 +7,7 @@
 	-I$(top_srcdir)/src/irc/core/ \
 	$(GLIB_CFLAGS)
 
-libirc_dcc_a_SOURCES = \
+libirssi_irc_dcc_la_SOURCES = \
 	dcc.c \
 	dcc-chat.c \
 	dcc-get.c \
diff -ur irssi-0.8.10-rc5-o/src/irc/flood/Makefile.am irssi-0.8.10-rc5/src/irc/flood/Makefile.am
--- irssi-0.8.10-rc5-o/src/irc/flood/Makefile.am	2004-05-10 14:45:04.000000000 -0600
+++ irssi-0.8.10-rc5/src/irc/flood/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libirc_flood.a
+libirssi_irc_flood_LTLIBRARIES = libirssi_irc_flood.la
+libirssi_irc_flooddir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -6,7 +7,7 @@
 	-I$(top_srcdir)/src/irc/core/ \
 	$(GLIB_CFLAGS)
 
-libirc_flood_a_SOURCES = \
+libirssi_irc_flood_la_SOURCES = \
 	autoignore.c \
 	flood.c
 
diff -ur irssi-0.8.10-rc5-o/src/irc/Makefile.am irssi-0.8.10-rc5/src/irc/Makefile.am
--- irssi-0.8.10-rc5-o/src/irc/Makefile.am	2002-03-15 14:07:19.000000000 -0700
+++ irssi-0.8.10-rc5/src/irc/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -4,9 +4,10 @@
 
 SUBDIRS = core dcc flood notifylist $(PROXY)
 
-noinst_LIBRARIES = libirc.a
+libirssi_irc_LTLIBRARIES = libirssi_irc.la
+libirssi_ircdir = $(libdir)
 
-libirc_a_SOURCES = irc.c
+libirssi_irc_la_SOURCES = irc.c
 
 DISTFILES = $(DIST_COMMON) $(HEADERS) $(TEXINFOS) $(EXTRA_DIST)
 
diff -ur irssi-0.8.10-rc5-o/src/irc/notifylist/Makefile.am irssi-0.8.10-rc5/src/irc/notifylist/Makefile.am
--- irssi-0.8.10-rc5-o/src/irc/notifylist/Makefile.am	2004-05-10 14:45:04.000000000 -0600
+++ irssi-0.8.10-rc5/src/irc/notifylist/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,4 +1,5 @@
-noinst_LIBRARIES = libirc_notifylist.a
+libirssi_irc_notifylist_LTLIBRARIES = libirssi_irc_notifylist.la
+libirssi_irc_notifylistdir = $(libdir)
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
@@ -6,7 +7,7 @@
 	-I$(top_srcdir)/src/irc/core/ \
 	$(GLIB_CFLAGS)
 
-libirc_notifylist_a_SOURCES = \
+libirssi_irc_notifylist_la_SOURCES = \
 	notifylist.c \
 	notify-commands.c \
 	notify-ison.c \
diff -ur irssi-0.8.10-rc5-o/src/lib-config/Makefile.am irssi-0.8.10-rc5/src/lib-config/Makefile.am
--- irssi-0.8.10-rc5-o/src/lib-config/Makefile.am	2004-05-10 14:45:04.000000000 -0600
+++ irssi-0.8.10-rc5/src/lib-config/Makefile.am	2005-05-29 01:32:38.000000000 -0600
@@ -1,10 +1,11 @@
-noinst_LIBRARIES = libirssi_config.a
+libirssi_config_LTLIBRARIES = libirssi_config.la
+libirssi_configdir = /usr/lib
 
 INCLUDES = \
 	-I$(top_srcdir)/src \
 	$(GLIB_CFLAGS)
 
-libirssi_config_a_SOURCES = \
+libirssi_config_la_SOURCES = \
 	get.c \
 	set.c \
 	parse.c \
diff -ur irssi-0.8.10-rc5-o/src/perl/Makefile.am irssi-0.8.10-rc5/src/perl/Makefile.am
--- irssi-0.8.10-rc5-o/src/perl/Makefile.am	2003-11-16 11:30:02.000000000 -0700
+++ irssi-0.8.10-rc5/src/perl/Makefile.am	2005-05-29 01:33:18.000000000 -0600
@@ -7,11 +7,11 @@
 module_LTLIBRARIES = $(perl_module_lib) $(perl_module_fe_lib)
 noinst_LTLIBRARIES = $(perl_static_lib) $(perl_static_fe_lib)
 EXTRA_LTLIBRARIES = \
-	libperl_core.la libfe_perl.la \
-	libperl_core_static.la libfe_perl_static.la
+	libirssi_perl_core.la libirssife_perl.la \
+	libirssi_perl_core_static.la libirssife_perl_static.la
 
-libperl_core_la_LDFLAGS = -avoid-version -rpath $(moduledir)
-libfe_perl_la_LDFLAGS = -avoid-version -rpath $(moduledir)
+libirssi_perl_core_la_LDFLAGS = -avoid-version -rpath $(moduledir)
+libirssife_perl_la_LDFLAGS = -avoid-version -rpath $(moduledir)
 
 perl-core.c: perl-signals-list.h irssi-core.pl.h
 
@@ -44,7 +44,7 @@
 	perl-signals.h \
 	perl-sources.h
 
-libperl_core_la_DEPENDENCIES = .libs/libperl_orig.a .libs/DynaLoader.a
+libirssi_perl_core_la_DEPENDENCIES = .libs/libperl_orig.a .libs/DynaLoader.a
 
 .libs/libperl_orig.a:
 	if [ ! -d .libs ]; then mkdir .libs; fi
@@ -55,16 +55,16 @@
 	rm -f .libs/DynaLoader.a
 	$(LN_S) $(DYNALOADER_A) .libs/DynaLoader.a
 
-libperl_core_la_SOURCES = \
+libirssi_perl_core_la_SOURCES = \
 	$(perl_sources)
 
-libperl_core_static_la_SOURCES = \
+libirssi_perl_core_static_la_SOURCES = \
 	$(perl_sources)
 
-libfe_perl_la_SOURCES = \
+libirssife_perl_la_SOURCES = \
 	$(perl_fe_sources)
 
-libfe_perl_static_la_SOURCES = \
+libirssife_perl_static_la_SOURCES = \
 	$(perl_fe_sources)
 
 perl-signals-list.h: $(top_srcdir)/docs/signals.txt $(srcdir)/get-signals.pl
@@ -165,4 +165,4 @@
 	  cd ..; \
 	done
 
-libperl_core_la_LIBADD = $(PERL_LDFLAGS)
+libirssi_perl_core_la_LIBADD = $(PERL_LDFLAGS)
