diff -ur irssi-0.8.1.orig/src/irc/core/channels-query.c irssi-0.8.1/src/irc/core/channels-query.c
--- irssi-0.8.1.orig/src/irc/core/channels-query.c	Fri Dec  7 23:10:11 2001
+++ irssi-0.8.1/src/irc/core/channels-query.c	Mon Mar 11 15:07:55 2002
@@ -364,7 +364,8 @@
 	/* Add channel to query lists */
 	if (!channel->no_modes)
 		query_add_channel(channel, CHANNEL_QUERY_MODE);
-	query_add_channel(channel, CHANNEL_QUERY_WHO);
+	if (settings_get_bool("channel_auto_who"))
+		query_add_channel(channel, CHANNEL_QUERY_WHO);
 	if (!channel->no_modes)
 		query_add_channel(channel, CHANNEL_QUERY_BMODE);
 
@@ -481,6 +482,7 @@
 void channels_query_init(void)
 {
 	settings_add_bool("misc", "channel_sync", TRUE);
+	settings_add_bool("misc", "channel_auto_who", TRUE);
 
 	signal_add("server connected", (SIGNAL_FUNC) sig_connected);
 	signal_add("server disconnected", (SIGNAL_FUNC) sig_disconnected);
