diff -ur irssi-0.8.10/irssi.conf irssi-0.8.10.new/irssi.conf
--- irssi-0.8.10/irssi.conf	2005-12-08 17:32:50.000000000 +0000
+++ irssi-0.8.10.new/irssi.conf	2005-12-11 14:33:31.237162000 +0000
@@ -8,6 +8,7 @@
   },
   { address = "irc.dal.net"; chatnet = "DALnet"; port = "6667"; }, 
   { address = "irc.quakenet.org"; chatnet = "QuakeNet"; port = "6667"; },
+  { address = "irc.pld-linux.org"; chatnet = "PLDNet"; port = "6667"; },
   { address = "silc.silcnet.org"; chatnet = "SILC"; port = "706"; }
 );
 
@@ -44,10 +45,19 @@
     max_whois = "30";
   };
   SILC = { type = "SILC"; };
+  PLDNet = {
+    type = "IRC";
+    max_kicks = "4";
+    max_modes = "4";
+    max_msgs = "5";
+    max_whois = "4";
+    max_query_chans = "5";
+  };
 };
 
 channels = (
   { name = "#irssi"; chatnet = "ircnet"; autojoin = "No"; },
+  { name = "#pldhelp"; chatnet = "PLDNet"; autojoin = "Yes"; },
   { name = "silc"; chatnet = "silc"; autojoin = "No"; }
 );
 
