diff -urNp -x '*.orig' irssi-1.2.2.org/src/core/misc.h irssi-1.2.2/src/core/misc.h
--- irssi-1.2.2.org/src/core/misc.h	2019-08-29 15:48:46.000000000 +0200
+++ irssi-1.2.2/src/core/misc.h	2021-05-04 21:22:59.014622440 +0200
@@ -4,6 +4,10 @@
 int g_input_add_poll(int fd, int priority, int condition,
 		     GInputFunction function, void *data);
 
+#ifdef HAVE_CONFIG_H
+#include "irssi-config.h"
+#endif
+
 /* `str' should be type char[MAX_INT_STRLEN] */
 #define ltoa(str, num) \
 	g_snprintf(str, sizeof(str), "%d", num)
diff -urNp -x '*.orig' irssi-1.2.2.org/src/perl/module.h irssi-1.2.2/src/perl/module.h
--- irssi-1.2.2.org/src/perl/module.h	2019-08-29 15:48:46.000000000 +0200
+++ irssi-1.2.2/src/perl/module.h	2021-05-04 21:22:59.017955947 +0200
@@ -1,5 +1,9 @@
 #include "common.h"
 
+#ifdef HAVE_CONFIG_H
+#include "irssi-config.h"
+#endif
+
 #ifdef NEED_PERL_H
 #  include <EXTERN.h>
 #  ifndef _SEM_SEMUN_UNDEFINED
