diff -Nru irssi-0.7.28/Makefile.am irssi-0.7.28.new/Makefile.am
--- irssi-0.7.28/Makefile.am	Sat Mar 11 19:40:28 2000
+++ irssi-0.7.28.new/Makefile.am	Sat Mar 11 22:35:21 2000
@@ -20,7 +20,7 @@
 @MAINT@	cd macros && $(MAKE) macros.dep
 
 if HAVE_GNOME
-desktopdir = $(datadir)/gnome/apps/Internet
+desktopdir = $(datadir)/applnk/Networking/IRC
 desktop_DATA = irssi.desktop
 
 corbadir = $(sysconfdir)/CORBA/servers
