--- irssi-0.8.11/src/core/misc.h	2007-04-25 20:13:56.000000000 +0200
+++ irssi-0.8.11/src/core/misc.h-new	2007-06-07 01:41:35.677176190 +0200
@@ -1,6 +1,10 @@
 #ifndef __MISC_H
 #define __MISC_H
 
+#ifdef HAVE_CONFIG_H
+#include "irssi-config.h"
+#endif
+
 /* `str' should be type char[MAX_INT_STRLEN] */
 #define ltoa(str, num) \
 	g_snprintf(str, sizeof(str), "%d", num)
--- irssi-0.8.11/src/perl/module.h	2007-04-25 20:13:56.000000000 +0200
+++ irssi-0.8.11/src/perl/module.h-new	2007-06-07 01:42:22.832959937 +0200
@@ -1,5 +1,9 @@
 #include "common.h"
 
+#ifdef HAVE_CONFIG_H
+#include "irssi-config.h"
+#endif
+
 #ifdef NEED_PERL_H
 #  include <EXTERN.h>
 #  ifndef _SEM_SEMUN_UNDEFINED
