--- irssi-0.8.11/curses.m4.orig	2007-06-07 01:22:01.709666033 +0200
+++ irssi-0.8.11/curses.m4	2007-06-07 01:24:43.928073732 +0200
@@ -232,17 +232,17 @@
 AC_DEFUN(AC_SEARCH_NCURSES, [
     AC_CHECKING("location of ncurses.h file")
 
-    AC_NCURSES(/usr/include, ncurses.h, -lncurses,,
+    AC_NCURSES(/usr/include, ncurses.h, -lncurses -ltinfo,,
 	       [ncurses in /usr/include])
-    AC_NCURSES(/usr/include/ncurses, ncurses.h, -lncurses, -I/usr/include/ncurses,
+    AC_NCURSES(/usr/include/ncurses, ncurses.h, -lncurses -ltinfo, -I/usr/include/ncurses,
 	       [ncurses in /usr/include/ncurses])
-    AC_NCURSES(/usr/local/include, ncurses.h, -L/usr/local/lib -lncurses, -I/usr/local/include,
+    AC_NCURSES(/usr/local/include, ncurses.h, -L/usr/local/lib -lncurses -ltinfo, -I/usr/local/include,
 	       [ncurses in /usr/local/include])
-    AC_NCURSES(/usr/local/include/ncurses, ncurses.h, -L/usr/local/lib -lncurses, -I/usr/local/include/ncurses,
+    AC_NCURSES(/usr/local/include/ncurses, ncurses.h, -L/usr/local/lib -lncurses -ltinfo, -I/usr/local/include/ncurses,
 	       [ncurses in /usr/local/include/ncurses])
-    AC_NCURSES(/usr/local/include/ncurses, curses.h, -L/usr/local/lib -lncurses, -I/usr/local/include/ncurses -DRENAMED_NCURSES,
+    AC_NCURSES(/usr/local/include/ncurses, curses.h, -L/usr/local/lib -lncurses -ltinfo, -I/usr/local/include/ncurses -DRENAMED_NCURSES,
 	       [renamed ncurses in /usr/local/include/ncurses])
-    AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses, -I/usr/include/ncurses -DRENAMED_NCURSES,
+    AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses -ltinfo, -I/usr/include/ncurses -DRENAMED_NCURSES,
 	       [renamed ncurses in /usr/include/ncurses])
 
     dnl
