diff -ur irssi-0.8.10-rc5.orig/src/irc/core/channels-query.c irssi-0.8.10-rc5/src/irc/core/channels-query.c
--- irssi-0.8.10-rc5.orig/src/irc/core/channels-query.c	2004-03-10 20:29:15.000000000 +0100
+++ irssi-0.8.10-rc5/src/irc/core/channels-query.c	2004-10-03 07:13:10.165114184 +0200
@@ -367,6 +367,8 @@
 	if (g_hash_table_size(channel->nicks) <
 	    settings_get_int("channel_max_who_sync"))
 		query_add_channel(channel, CHANNEL_QUERY_WHO);
+	if (settings_get_bool("channel_auto_who"))
+		query_add_channel(channel, CHANNEL_QUERY_WHO);
 	if (!channel->no_modes)
 		query_add_channel(channel, CHANNEL_QUERY_BMODE);
 
@@ -484,6 +486,7 @@
 {
 	settings_add_bool("misc", "channel_sync", TRUE);
 	settings_add_int("misc", "channel_max_who_sync", 1000);
+	settings_add_bool("misc", "channel_auto_who", TRUE);
 
 	signal_add("server connected", (SIGNAL_FUNC) sig_connected);
 	signal_add("server disconnected", (SIGNAL_FUNC) sig_disconnected);
