--- irssi-0.8.9/irssi.conf	2004-01-20 00:43:34.701543232 +0000
+++ irssi-0.8.9/irssi.conf.orig	2004-01-20 00:43:37.746080392 +0000
@@ -22,6 +22,7 @@
   },
   { address = "irc.ptnet.org"; chatnet = "PTnet"; port = "6667"; },
   { address = "irc.azzurra.org"; chatnet = "AzzurraNET"; port = "6667"; },
+  { address = "irc.pld-linux.org"; chatnet = "PLDNet"; port = "6667"; },
   { address = "silc.silcnet.org"; chatnet = "SILC"; port = "706"; }
 );
 
@@ -104,11 +105,20 @@
     max_msgs = "3";
   };
   SILC = { type = "SILC"; };
+  PLDNet = {
+    type = "IRC";
+    max_kicks = "4";
+    max_modes = "4";
+    max_msgs = "5";
+    max_whois = "4";
+    max_query_chans = "5";
+  };
 };
 
 channels = (
   { name = "#irssi"; chatnet = "ircnet"; autojoin = "No"; },
   { name = "#irssi"; chatnet = "freenode"; autojoin = "No"; },
+  { name = "#pldhelp"; chatnet = "PLDNet"; autojoin = "Yes"; },
   { name = "silc"; chatnet = "silc"; autojoin = "No"; }
 );
 
