diff -urNp -x '*.orig' irssi-1.2.2.org/configure.ac irssi-1.2.2/configure.ac
--- irssi-1.2.2.org/configure.ac	2021-05-04 21:23:37.126601768 +0200
+++ irssi-1.2.2/configure.ac	2021-05-04 21:23:37.263275533 +0200
@@ -431,7 +431,7 @@ if test "$want_perl" != "no"; then
 		AC_MSG_RESULT([not found, building without Perl])
 		want_perl=no
 	else
-		PERL_LDFLAGS=`$perlpath -MExtUtils::Embed -e ldopts 2>/dev/null`
+		PERL_LDFLAGS=`$perlpath -MExtUtils::Embed -e ldopts 2>/dev/null | sed -e 's/-Wl,--as-needed//'`
 
 		dnl * remove all database stuffs
 		dnl * nsl is already in ldflags
